### PR TITLE
Feature/Reworked main tabs (#2952)

### LINKED ISSUE #2952: Rework tabs
[comment]: # (Please replace ... with your information. Remove < and >)

### Description

[comment]: # (Feature or Bug? i.e Type: Bug)
*Type*: Feature

remove unused tabs (contacts, discovery), remove drawer (profile sidebar), prepare for home tab update, remove tab swipe UI

### PR DESCRIPTION
**Pull request by flexsurfer**
fixes #2952

removed swiper and drawer npm libs

### CODE REVIEW

File: src/status_im/chat/views/toolbar_content.cljs [hunk 1]
```diff
@@ -4,8 +4,8 @@
             [clojure.string :as str]
             [cljs-time.core :as t]
             [status-im.ui.components.react :refer [view
-                                                text
```
Comment (jeluard) [inline src/status_im/chat/views/toolbar_content.cljs:7]: Maybe we could use namespace here?

File: src/status_im/ui/components/icons/vector_icons.cljs [hunk 2]
```diff
@@ -23,12 +23,17 @@
...
 (def defs (get-class "Defs"))
 
-(def icons {:icons/chats               (slurp-svg "./resources/icons/bottom/chats_gray.svg")
-            :icons/chats-active        (slurp-svg "./resources/icons/bottom/chats_active.svg")
-            :icons/contacts            (slurp-svg "./resources/icons/bottom/contacts_gray.svg")
-            :icons/contacts-active     (slurp-svg "./resources/icons/bottom/contacts_active.svg")
-            :icons/discover            (slurp-svg "./resources/icons/bottom/discover_gray.svg")
-            :icons/discover-active     (slurp-svg "./resources/icons/bottom/discover_active.svg")
+(def icons {;;NOT USED IN MVP
+            ;;:icons/chats               (slurp-svg "./resources/icons/bottom/chats_gray.svg")
```
Comment (jeluard) [inline src/status_im/ui/components/icons/vector_icons.cljs]: How about deleting comments? They will still be accessible via git history if needed.

**Review by maxhora — APPROVED** [review]

File: src/status_im/ui/screens/main_tabs/views.cljs [hunk 3]
```diff
@@ -1,58 +1,65 @@
...
-            [status-im.ui.screens.wallet.main.views :refer [wallet]])
-  (:require-macros [status-im.utils.views :as views]))
+            [status-im.ui.screens.chats-list.views :as chats-list.views]
+            [status-im.ui.screens.profile.views :as profile.views]
+            [status-im.ui.screens.wallet.main.views :as wallet.views]
+            [status-im.ui.screens.main-tabs.styles :as styles]
+            [re-frame.core :as re-frame]))
+
+(def tabs-list-data
+  {:chat-list
```
Comment (jeluard) [inline src/status_im/ui/screens/main_tabs/views.cljs]: Any particular reason for this change? Also `view-id` can probably be removed and replaced by the key value.

File: src/status_im/ui/screens/views.cljs [hunk 4]
```diff
@@ -73,7 +72,7 @@
     (when view-id
       (let [current-view (validate-current-view view-id signed-up?)]
         (let [component (case current-view
-                          (:wallet :chat-list :discover :contact-list) main-tabs
+                          (:wallet :chat-list :my-profile) main-tabs
```
Comment (jeluard) [inline src/status_im/ui/screens/views.cljs]: Shall we rename `:chat-list` to `:home` for clarity?

**Review by jeluard — COMMENTED** [review]
Great! Lots of things removed!

File: src/status_im/chat/views/toolbar_content.cljs (hunk 1 above)
Comment (flexsurfer) [reply src/status_im/chat/views/toolbar_content.cljs:7]: it's just cursive, not related to this PR

**Review by janherich — APPROVED** [review]

File: src/status_im/ui/components/icons/vector_icons.cljs (hunk 2 above)
Comment (flexsurfer) [reply src/status_im/ui/components/icons/vector_icons.cljs]: ok

File: src/status_im/chat/views/toolbar_content.cljs (hunk 1 above)
Comment (jeluard) [reply src/status_im/chat/views/toolbar_content.cljs:7]: I know but it's an opportunity to move to `:as react` :)

File: src/status_im/ui/screens/views.cljs (hunk 4 above)
Comment (flexsurfer) [reply src/status_im/ui/screens/views.cljs]: hm, good question, decided to don't do this in this PR, but maybe you're right better to rename for clarity

File: src/status_im/ui/screens/main_tabs/views.cljs (hunk 3 above)
Comment (flexsurfer) [reply src/status_im/ui/screens/main_tabs/views.cljs]: what change do you mean? changed vector to map? yeah you're right

File: src/status_im/chat/views/toolbar_content.cljs (hunk 1 above)
Comment (flexsurfer) [reply src/status_im/chat/views/toolbar_content.cljs:7]: okay

Comment (flexsurfer) [reply src/status_im/chat/views/toolbar_content.cljs:7]: there are not only react, don't want to mess it in this PR

**Comment by flexsurfer** [comment]
@jeluard  done

**Review by jeluard — APPROVED** [review]

File: src/status_im/ui/screens/home/views.cljs [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+  (views/letsubs [chats        [:filtered-chats]
+                  edit?        [:get-in [:chat-list-ui-props :edit?]]
+                  show-search  [:get-in [:toolbar-search :show]]
+                  tabs-hidden? [:tabs-hidden?]]
+    [react/view styles/chats-container
+     [toolbar-view]
+     [list/flat-list {:style           styles/list-container
+                      :data            chats
+                      :render-fn       (fn [chat] [chat-list-item chat edit?])
+                      :header          (when-not (empty? chats) list/default-header)
```
Comment (rasom) [inline src/status_im/ui/screens/home/views.cljs:49]: https://github.com/bbatsov/clojure-style-guide#nil-punning

**Review by rasom — APPROVED** [review]

**Comment by asemiankevich** [comment]
Branch: PR-2992
Android: https://i.diawi.com/yf8X1W
iOS: https://i.diawi.com/7hKzRq

- no contacts and discovery tabs
- no drawer
- no tab swiping

Basic functionality works as expected, so I can add contact, faucet funds and send. No more testing was performed as this is only UI change.

**Pull request merged**